### PR TITLE
chore(deps): update dependencies

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Other

**Required reviews:** 
- 1

**What this does:**
- Updates dependencies
- Updates TypeScript to v5.2.2

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -18,8 +18,7 @@
     "plugin:eslint-comments/recommended",
     "plugin:@typescript-eslint/recommended",
     "plugin:import/typescript",
-    "prettier",
-    "prettier/@typescript-eslint"
```
Comment (manumonti) [inline .eslintrc.json:22]: Could you please elaborate on why "prettier/@typescript-eslint" is no more needed in .eslitrc?

**Review by manumonti — APPROVED** [review]
LGTM 👌! I left a question just out of curiosity

File: .eslintrc.json (hunk 1 above)
Comment (piotr-roslaniec) [reply .eslintrc.json:22]: It's now part of `@typescript-eslint/recommended`

File: types/ethers-contracts/index.ts [hunk 2]
```diff
@@ -3,6 +3,6 @@
 /* eslint-disable */
 export type { Coordinator } from "./Coordinator";
 export type { SubscriptionManager } from "./SubscriptionManager";
-
+export * as factories from "./factories";
```
Comment (derekpierre) [inline types/ethers-contracts/index.ts:6]: This file is autogenerated?

**Review by derekpierre — APPROVED** [review]
🎸  - just one comment about something I'm curious about. Looks good otherwise.

File: types/ethers-contracts/index.ts (hunk 2 above)
Comment (piotr-roslaniec) [reply types/ethers-contracts/index.ts:6]: Yes, it's generated by `typechain`.

**Pull request merged**